### PR TITLE
RDS Core: separate general methods from the sriov verification suite

### PR DESCRIPTION
**Pull request by elenagerman**
- helper.go file created

- verifySRIOVConnectivity was renamed to the verifyConnectivity and moved to the helper.go

- createServiceAccount, deleteServiceAccount, deleteClusterRBAC, createClusterRBAC, deleteConfigMap, createConfigMap, deleteDeployments, findPodWithSelector, defineContainer, verifyMsgInPodLogs were moved to the helper.go

- 'if pState == "NonCompliant"' state was changed to the 'if pState == "NonCompliant"'. Depends on deployment type it can be some extra policies with no state at all be found and it can fail test with no reason

### CODE REVIEW

**Comment by elenagerman** [comment]
@yprokule Hi! as was discussed, PTAL

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+)
+
+func createServiceAccount(saName, nsName string) {
+	ginkgo.By(fmt.Sprintf("Creating ServiceAccount %q in %q namespace",
+		saName, nsName))
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Creating SA %q in %q namespace",
+		saName, nsName)
+
+	insureNamespaceExists(nsName)
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:32]: this shouldn't be part of the method, it should only create SA, _namespace_ management should be done outside

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+		}).WithContext(ctx).WithPolling(5*time.Second).WithTimeout(1*time.Minute).Should(gomega.BeTrue(),
+			"Failed to delete configMap")
+	}
+}
+
+func createConfigMap(cmName, nsName string, data map[string]string) {
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Create ConfigMap %q in %q namespace",
+		cmName, nsName)
+
+	insureNamespaceExists(nsName)
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:217]: ditto

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Logs from pod %s:\n%s", podObj.Definition.Name, podLog)
+
+		return true
+	}).WithContext(ctx).WithPolling(5*time.Second).WithTimeout(1*time.Minute).Should(gomega.BeTrue(),
+		fmt.Sprintf("Failed to get logs from pod %q", podObj.Definition.Name))
+
+	gomega.Expect(podLog).Should(gomega.ContainSubstring(msg))
+}
+
+func insureNamespaceExists(nsName string) {
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:414]: I assume it should be `ensure` and not `insure` 😸

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+
+	if !deploySa.Exists() {
+		var ctx ginkgo.SpecContext
+
+		deploySa, err := deploySa.Create()
+
+		if err != nil {
+			glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Error creating SA %q in %q namespace: %v",
+				saName, nsName, err)
+		}
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:41]: This should be inside `Eventually` block

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+func createServiceAccount(saName, nsName string) {
+	ginkgo.By(fmt.Sprintf("Creating ServiceAccount %q in %q namespace",
+		saName, nsName))
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Creating SA %q in %q namespace",
+		saName, nsName)
+
+	deploySa := serviceaccount.NewBuilder(rdscoreinittools.APIClient, saName, nsName)
+
+	if !deploySa.Exists() {
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:33]: please add logging for a case that `SA` already exists, thanks

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+			saName, nsName)
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Deleting ServiceAccount %q in %q namespace",
+			saName, nsName)
+
+		err := deploySa.Delete()
+
+		if err != nil {
+			glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Error deleting ServiceAccount %q in %q namespace: %v",
+				saName, nsName, err)
+		}
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:79]: This should be within `Eventually` block

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 7]
```diff
@@ -0,0 +1,471 @@
...
+		rdscoreinittools.APIClient,
+		rbacName); err == nil {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof("ClusterRoleBinding %q found. Deleting...", rbacName)
+
+		err := crbSa.Delete()
+
+		if err != nil {
+			glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Error deleting ClusterRoleBinding %q : %v",
+				rbacName, err)
+		}
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:118]: ditto, should be within `Eventually` block

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 8]
```diff
@@ -0,0 +1,471 @@
...
+			Name:      saName,
+			Kind:      "ServiceAccount",
+			Namespace: nsName,
+		})
+
+	crbSa, err := crbSa.Create()
+	if err != nil {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof(
+			"Error Creating ClusterRoleBinding %q : %v", crbSa.Definition.Name, err)
+	}
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:157]: ditto, should be within `Eventually` block

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 9]
```diff
@@ -0,0 +1,471 @@
...
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof("configMap %q found, deleting", cmName)
+
+		var ctx ginkgo.SpecContext
+
+		err := cmBuilder.Delete()
+
+		if err != nil {
+			glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Error deleting configMap %q : %v",
+				cmName, err)
+		}
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:190]: ditto

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 10]
```diff
@@ -0,0 +1,471 @@
...
+	cmBuilder.WithData(data)
+
+	var ctx ginkgo.SpecContext
+
+	cmResult, err := cmBuilder.Create()
+	if err != nil {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Error creating ConfigMap %q in %q namespace: %v",
+			cmName, nsName, err)
+	}
+
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:222]: ditto

File: tests/system-tests/rdscore/internal/rdscorecommon/helper.go [hunk 11]
```diff
@@ -0,0 +1,471 @@
...
+}
+
+func deleteDeployments(dName, nsName string) {
+	ginkgo.By(fmt.Sprintf("Removing test deployment %q from %q ns", dName, nsName))
+
+	if deploy, err := deployment.Pull(rdscoreinittools.APIClient, dName, nsName); err == nil {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Deleting deployment %q from %q namespace",
+			deploy.Definition.Name, nsName)
+
+		deploy, err = deploy.WithReplicas(int32(0)).Update()
```
Comment (yprokule) [inline tests/system-tests/rdscore/internal/rdscorecommon/helper.go:247]: no need to scale down deployment before deletion

**Review by yprokule — CHANGES_REQUESTED** [review]
please address new comments, thanks

**Comment by elenagerman** [comment]
@yprokule I'm sorry, but I do not understand why general methods like delete/create API objects cannot be outside the sriov validation test. What if we need it for the other operator's tests? For example, configmap, rbac, and sa config can be required for the keda config.
I also do not understand what is terrible in double-checking if ns exists.
Anyway, with all the requested changes, we are returning to the original version; I do not see any value in this MR.
Thank you!

**Comment by yprokule** [comment]
> @yprokule I'm sorry, but I do not understand why general methods like delete/create API objects cannot be outside the sriov validation test. What if we need it for the other operator's tests? For example, configmap, rbac, and sa config can be required for the keda config.

@elenagerman what are U talking about? I didn't object about having some of the common methods in a dedicated file. What I said is that during your _move_ of existing methods to a new file U broke the existing logic and were asked to fix it

> I also do not understand what is terrible in double-checking if ns exists.

U don't double check but create if it's missing which again affects existing logic. And what the problem to run this at the beginning of the test or in the setup block?

> Anyway, with all the requested changes, we are returning to the original version; I do not see any value in this MR.

Again, not sure where U got the idea that we are returning to the original version - all U had to do was to address broken logic in the PR.

Feel free to reopen PR and address comments or I might piggyback on this work and adjust it myself.

Thank you!

**Pull request closed without merging**